### DISCUSSION TITLE
Other unsuported regions

### DISCUSSION THREAD

**Discussion by arvvoid** [arvvoid/plugin.video.hbogoeu#204] [Ideas]
HBO Go US (http://hbogo.com) [NOT SUPPORTED BY THIS ADD-ON]
HBO Go Asia (https://www.hbogoasia.com/) [NOT SUPPORTED BY THIS ADD-ON]
HBO Go Latin America (https://www.hbogo.com.br/) [NOT SUPPORTED BY THIS ADD-ON]

If more details become known or a working add-on for those regions become available. Inclusion might be possible and a unified Hbogo Kodi add-on might be possible.

#6 #7

**Comment by bal8784** [comment]
How can I help with HBO Go US? Ive found some hosts that get hit when accessing hbogo.

comet.api.hbo.com  (this is coded into the android apk in CometClient.java)
login.api.hbo.com
markers.api.hbo.com
hbobroadband.122.2o7.net
artist.api.lv3.cdn.hbo.com
auth.hbogo.com
telegraph.api.hbo.com

**Comment by arvvoid** [comment]
Might work on that in the future, don't have the time now.

But contributions and pull requests are welcome :)

**Comment by MythodeaLoL** [comment]
I would be very grateful when you have the time to create an authentication module for HBO BR. I can help with login if u need. When you have time, call me. @arvvoid See u.

**Comment by arvvoid** [comment]
@FloridaStream Hello, when I'm ready and have more free time to work on it will let you know. 
It's not only a matter of solving the login, the API is different so login, device registration, operator listing, content listing, purchase, search, my list, ecc... all have to be figured out. If someone figures out the calls necessary and responses implementing it in the add-on is not a problem.

**Comment by arvvoid** [comment]
If someone is willing to do a browser log performing all the actions on the hbo go website with chrome developer tools and post the log I can start evalueting what changes have to be made (be informed that the log will probably contain your credentials so remove them before sending the log with a text editor search and replace function, and change your password after taking the log):

1) Go and login on Hbo website and Open the Developer Tools

![Screenshot 2019-11-04 at 09 30 31](https://user-images.githubusercontent.com/46710439/68108052-fbaf1780-fee6-11e9-83bc-0de1eb28a097.png)
 
2) Open the Network TAB
3) Check the preserve log checkbox, login, perform all these actionss, and also play a video for few minutes so i have actions around playback logged as well.

5) After you performed all these actions save the log (right-clik on the list to show the context menu)

![Screenshot 2019-11-04 at 09 32 07](https://user-images.githubusercontent.com/46710439/68108050-fb168100-fee6-11e9-8360-66bc51d9d11c.png)

6) Open the log with a text editor replace all occurrences of your e-mail with XXX and replace all occurrences of your password with XXXXX using the search and replace function of your text editor. Save and send the log.

@FloridaStream @bal8784

**Comment by HereInPlainSight** [comment]
Hiya!  I have HBO Go in the US, with the provider Optimum.  You didn't specify exactly what 'these actions' are in the 'perform these actions' part, so...  I did everything.  Went to all the different categories and subsections.  If you need a smaller log with only specific actions (like just logging in through my provider and then playing a video), please let me know.

Due to the size of the log file, I couldn't upload it here, I had to toss it up on [google drive](https://drive.google.com/open?id=1_qi1gdHoXVadVEtKwEVfhajPVkY9OjqZ).  I made substitutions as 'USER_EMAIL', 'USER_NAME' (what I signed in with, USER_EMAIL is just USER_NAME@optimum.net, which makes me question why I bothered to substitute it at all now that I think about it), and 'USER_PASSWORD' as appropriate.

My do-goodish nature is largely variable on how busy things are, so if you need something else and I take a while to get it, I apologize.

**Comment by arvvoid** [comment]
@HereInPlainSight Hello, thanks for sharing the log. It is a starting point. Going to analyze it for the weekend most likely if I have time. From a quick look, the API is completely different so a new handler for US region has to be written from 0. 
The log is ok, since we are in 100% uncharted territory more data and more actions covered the better.
When I have some time will try to do a tentative prototype, but since I cant interactively test this and experiment with it, without the active collaboration of a local developer with direct access there is a very slim chance of a working version.

I downloaded the log so if you want to remove it it's safe to do.

**Comment by nomad-fr** [comment]
I'd like to help about ocs.fr support.
For the moment I've no subscribtion.
In first I'd like to know if the 'smallest' subscription is enough : [https://www.ocs.fr/souscription/selection](https://www.ocs.fr/souscription/selection)

**Comment by arvvoid** [comment]
@nomad-fr Hello, this does not look like anything similar to the HBO Go EU service. Probably that service is better suited in a separate dedicated add-on.
Since I'm not familiar with the service and have no access to it I can't provide any directions.
